slow_odgi/flip.py

```diff
@@ -16,21 +17,55 @@ def path_is_rev(path, graph):
 
 
 def flip_path(path, graph):
-    """Flip the given path, if it is more reverse-oriented than forward."""
+    """Flip the given path if it is more reverse- than forward-oriented."""
     if path_is_rev(path, graph):
-        segments = []
+        path_segs = []
         for seg in reversed(path.segments):
-            segments.append (mygfa.Handle(seg.name, not seg.orientation))
-        return mygfa.Path(path.name+"_inv", segments, path.overlaps)
+            path_segs.append(mygfa.Handle(seg.name, not seg.orientation))
+        return mygfa.Path(f"{path.name}_inv", path_segs, path.overlaps)
     else:
         return path
 
 
+def dedup(list):
+    new = []
+    for item in list:
+        if item not in new and item.rev() not in new:
+            # odgi seems to consider a link's reverse its own duplicate.
+            new.append(item)
+    return new
+
+
+def gen_links(paths, prop) -> List[mygfa.Link]:
+    """Given a list of paths and a proposition on paths,
```

Comment:
I chucked at calling this `pop` (your theorem-prover background taught you well!). No need for any changes, but I might have called this a "predicate" (i.e., a function from something to bool).